packages/vite/src/node/utils.ts

```diff
@@ -433,10 +433,8 @@ export function posToNumber(
   const lines = source.split(splitRE)
   const { line, column } = pos
   let start = 0
-  for (let i = 0; i < line - 1; i++) {
-    if (lines[i]) {
-      start += lines[i].length + 1
-    }
```

Comment:
This condition was added because of a reported issue https://github.com/vitejs/vite/issues/5647#issuecomment-997070908. But it looks like this guard isn't correct, if `lines[i]` is `''` then a +1 is needed here.